lib/ethernet/server.ex

```diff
@@ -49,25 +49,28 @@ defmodule Nerves.IO.Ethernet.Server do
   # try renewing dhcp lease upon expiration unless we've been configured
   # as a static ip in the meantime
   def handle_info(:dhcp_lease_expired, state) do
-    case state.status do
-      "static" -> state
-      _ -> configure_with_dynamic_ip(state)
-    end
-    |> respond :noreply
+    state.status
+    |> renew_dhcp(state)
+    |> respond(:noreply)
   end
 
+  defp renew_dhcp("static", state), do: state
+  defp renew_dhcp(_, state), do: configure_with_dynamic_ip(state)
+
   # called periodically to try to see if a dhcp server came back
   # online
   def handle_info(:ip4ll_dhcp_retry, state) do
```

Comment:
@ghitchens Please, move this function before `renew_dhcp`, probably you're a receiving a warning right now.